Added TOR connection routing

whiterabb17 opened the pull request:
Nothing major changed, a few additional settings to handle TOR (namely torPort & torRoute)

Added a small visual change to Port settings (setting port text color to orange if that port is routing TOR)

Server will auto download and spinup tor if using
Client will auto download and spinup tor if using

All features tested and confirmed working under TOR connection. `Stubs` folder now contains client.xml & clientTor.xml specifying additional dlls needed for TOR - but to ensure no additional weight to payload if not using TOR

Initial callback to the server from the client is not instant [ obviously due to needing to get required tor binaries, spinup the tor proxy and connect ] - but once connected there does not seem to be any reduced performance for any features ( remote desktop potentially small lag depending on connection speeds both ends ) 

Small bug if client cannot reach server while using TOR, as the client uses host chains - the client tries to invoke a new privoxy instance instead of using existing or killing existing and restarting - Extremely easy to fix, just as i have no more time right now, i will fix end of my day.

Should you not want to integrate this pull then no worries. :)

whiterabb17 committed:
Added TOR connection routing

Nothing major changed, a few additional settings to handle TOR (namely torPort & torRoute)

Added a small visual change to Port settings (setting port text color to orange if that port is routing TOR)

Server will auto download and spinup tor if using
Client will auto download and spinup tor if using

All features tested and confirmed working under TOR connection.
`Stubs` folder now contains client.xml & clientTor.xml specifying additional dlls needed for TOR - but to ensure no additional weight to payload if not using TOR

Small bug if client cannot reach server while using TOR, as the client uses host chains - the client tries to invoke a new privoxy instance instead of using existing or killing existing and restarting - Extremely easy to fix, just as i have no more time right now, i will fix end of my day.

Should you not want to integrate this pull then no worries. :)

arsium commented:
> Nothing major changed, a few additional settings to handle TOR (namely torPort & torRoute)
> 
> Added a small visual change to Port settings (setting port text color to orange if that port is routing TOR)
> 
> Server will auto download and spinup tor if using Client will auto download and spinup tor if using
> 
> All features tested and confirmed working under TOR connection. `Stubs` folder now contains client.xml & clientTor.xml specifying additional dlls needed for TOR - but to ensure no additional weight to payload if not using TOR
> 
> Initial callback to the server from the client is not instant [ obviously due to needing to get required tor binaries, spinup the tor proxy and connect ] - but once connected there does not seem to be any reduced performance for any features ( remote desktop potentially small lag depending on connection speeds both ends )
> 
> Small bug if client cannot reach server while using TOR, as the client uses host chains - the client tries to invoke a new privoxy instance instead of using existing or killing existing and restarting - Extremely easy to fix, just as i have no more time right now, i will fix end of my day.
> 
> Should you not want to integrate this pull then no worries. :)

Really good thing ! I will add it as soon as I check it :) (won't be merged because I've already begun working on next update so  it will be downloaded locally and added manually)
The client is not too heavy with tor libraries ?

whiterabb17 commented:
Final built Client including your pre-build obf is 200kb

whiterabb17 commented:
In Server folder DLLs (OraclesBridge.dll) is server only
In c2 torLibs (OraclesBridge.dll) is client only

whiterabb17 committed:
updated given Release Stub files

Updated bin\Release\Stubs to hold up to date libs, xmls & client with proxify fix
(Checking if awake before trying to spin up the proxy fixes the restarting of privoxy)

arsium commented:
Also I don't know how to setup tor server to use it with Eagle (never used tor in project or personally)

whiterabb17 committed:
Multiple Changes

If client selects the TOR port in builder, onion address is added to Cell[0], if hostList is null due to only tor added as a callback, then 127.0.0.1:<torPort> is added as hostList value to avoid issues during compilation.

Added obfuscating values for all new functions during built.
Fixed multiple privoxy on fail bug
Gave end-user option to allow proxifying post-infection, should the togglebox be checked, will build with TOR config building in the libs and keeping in proxy code.
If not checked, regular build is used with no proxy functions added to the actual build.
Fixed conflicting `system.net.http` lib warn during build
Coded logic that it wont automatically add tor things if the user has TOR on, now the togglebox must be checked for the TOR client to build - ensuring the user has the option to still build normal clients (while TOR is enabled)

whiterabb17 commented:
Made some significant changes just for TOR building

* Also you dont need to do anything in terms of TOR
* All you need to do to use is start Eagle, goto settings and check use TOR and then set a listening port (Input done with InputBox after selecting <AddPort>) then restart.
* On the start with TOR enabled the proxy server is automatically spunup from Eagle setting all required settings.
* For the client, it must just be selected during build and when client starts proxy is automatically started (hidden) & configured then client just connects.

The addition requires absolutely no work/setup other then checking a box


CHANGES:
=========
If client selects the TOR port in builder, onion address is added to Cell[0], if hostList is null due to only tor added as a callback, then 127.0.0.1:<torPort> is added as hostList value to avoid issues during compilation.

Added obfuscating values for all new functions during built.
Fixed multiple privoxy on fail bug
Gave end-user option to allow proxifying post-infection, should the togglebox be checked, will build with TOR config building in the libs and keeping in proxy code.
If not checked, regular build is used with no proxy functions added to the actual build. 
Fixed conflicting `system.net.http` lib warn during build
Coded logic that it wont automatically add tor things if the user has TOR on, now the togglebox must be checked for the TOR client to build - ensuring the user has the option to still build normal clients (while TOR is enabled)

whiterabb17 commented:
DEMO
======
Demo of auto setup server-side
Building of client (update reduced tor build size from 200kb -> 164kb)

![tor](https://user-images.githubusercontent.com/90134688/197350785-9f387a0d-f46d-4457-81de-9755af71dbc8.gif)

whiterabb17 commented:
But again should you not want to integrate that's okay, just let me know and I'll remove the repo

vkyxah3 commented:
Please change GUI . Add multiple client operation at one time.... Now it does select 1 client only

arsium commented:
> But again should you not want to integrate that's okay, just let me know and I'll remove the repo

I think I won't include it but you could let the repository to show that it's easy to custom it :)

arsium commented:
> Please change GUI . Add multiple client operation at one time.... Now it does select 1 client only

This will come in next update.

NaderHana commented:
The exe payload end its task when i close the Eagle monitor, thats why i lose all my clients when i close it. Why's that and Can you Update it?

arsium commented:
> The exe payload end its task when i close the Eagle monitor, thats why i lose all my clients when i close it. Why's that and Can you Update it?

Debug it and you will probably have the answer :)

whiterabb17 commented:
I dont want to keep to many repos, so im going to remove it in the next hour. But your always welcome to get in contact should you want it implemented in the future and in that case i'll just recode it - took me all of 30mins or so

whiterabb17 commented:
> > The exe payload end its task when i close the Eagle monitor, thats why i lose all my clients when i close it. Why's that and Can you Update it?
> 
> Debug it and you will probably have the answer :)

Majority of these people have no idea what debugging is, let alone how to do it and fix an issue sadly

NaderHana commented:
thx but do mean get in visual basic 2022 and debug it ?

arsium commented:
> I dont want to keep to many repos, so im going to remove it in the next hour. But your always welcome to get in contact should you want it implemented in the future and in that case i'll just recode it - took me all of 30mins or so

Ok ok thx :)

arsium commented:
> > > The exe payload end its task when i close the Eagle monitor, thats why i lose all my clients when i close it. Why's that and Can you Update it?
> > 
> > Debug it and you will probably have the answer :)
> 
> Majority of these people have no idea what debugging is, let alone how to do it and fix an issue sadly

This is one the reason I removed Issues section

NaderHana commented:
bro i debug it and open it from bin it's the same when i close it i lose all clients

mandoof1 commented:
routing the network through tor seems smart but i dont understand how tor network can be port forwarded

whiterabb17 commented:
you dont port forward through TOR.
Which is why its appealing, dont need any public server.
But the developer said they did not want to integrate my pull of TOR networking. 
So there isnt any

mandoof1 commented:
> you dont port forward through TOR. Which is why its appealing, dont need any public server. But the developer said they did not want to integrate my pull of TOR networking. So there isnt any

yea i really liked the idea i dont know much c# but seeing what you did peaks my intrest i might study c# and replicate what you did

whiterabb17 commented:
Have fun, its not hard at all.
Though 2 dlls were my custom libraries

mandoof1 commented:
> Have fun, its not hard at all. Though 2 dlls were my custom libraries

dumb question when using the tor network is the dns static if not how do u get a static dns or is it a one time use thingy

whiterabb17 commented:
don't need static dns or static ip. you get an onion address. 
you have alot of reading to do to understand if your asking these questions
